prog/vm/host_nexus.rb

```diff
@@ -1,20 +1,33 @@
 # frozen_string_literal: true
 
 class Prog::Vm::HostNexus < Prog::Base
-  def self.assemble(sshable_hostname, location: "hetzner-hel1")
+  subject_is :vm_host
+
+  def self.assemble(sshable_hostname, provider: nil, location: "hetzner-hel1", hetzner_server_number: nil)
     DB.transaction do
       sa = Sshable.create(host: sshable_hostname)
-      VmHost.create(location: location) { _1.id = sa.id }
-
+      VmHost.create(location: location, provider: provider) { _1.id = sa.id }
+      HetznerHost.create(server_number: hetzner_server_number) { _1.id = sa.id } if provider == "hetzner"
```

Comment:
I saw `"hetzner"` string several time. It might be a good candidate for a constant.

Reply:
I'd be careful how I make the constant,  it'd be nice if printing it out was homoiconic, e.g.:

```
[2] clover-development(main)> module Provider
[2] clover-development(main)*   module Hetzner
[2] clover-development(main)*   end  
[2] clover-development(main)* end  
=> nil
[3] clover-development(main)> Provider::Hetzner
=> Provider::Hetzner
```

Example of something less useful:

```
[1] clover-development(main)> module Provider
[1] clover-development(main)*   Hetzner = 'hetzner'
[1] clover-development(main)* end  
=> "hetzner"
[2] clover-development(main)> Provider::Hetzner
=> "hetzner"
```

The reason being, you will always see the result of evaluation should you read any representation of data including the value.

Finally, I have found practical problems trying to avoid some kind of data literal for these kinds of things, in that the amount of representation swapping with text out of database can make it a cure worse than the disease. It also tends to prove strange if you serialize or de-serialize JSON.  Consider this problem, using the first variant of a `Provider::Hetzner` module:

```
puts JSON.generate({provider: Provider::Hetzner})
{"provider":"Provider::Hetzner"}
```

And here's what happens in a round-trip between generating and parsing:

```
puts JSON.parse(JSON.generate({provider: Provider::Hetzner}))
{"provider"=>"Provider::Hetzner"}
```

Now...what are you going to do with that string? Write `case` statements in one or even a few places to convert it into a `Provider::Hetzner` value?  Use eval? No, don't do that, it's a security vulnerability! How about `const_get`, as we are more or less obliged to use for Progs to match database record contents to code? I guess you can do that. But you still have to watch out for what you are const-getting. It's for this reason that `Prog::`s are boxed in their own module, to make it a little harder to wiggle an arbitrary code execution into the database, rather than const_getting absolutely anything. We could probably tighten it up even more.

But by the time you're done, was it worth it?

That's a case-by-case kind of thing, and there are ways you could mitigate it...also costing a bit more code.

A symbol is also a bit more idiomatic than a string for this kind of thing.

Reply:
Seems fine, I mean, I guess interactively I write "hetzner" but in source I'll write the symbol. Most of the time. Easier to grep I guess?